packages/react/src/components/List/ListItem.tsx

```diff
@@ -189,16 +210,11 @@ class ListItem extends UIComponent<ReactProps<ListItemProps>, ListItemState> {
         {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
       >
         {mediaElement}
+
         <Flex.Item grow>
-          <Flex column className={ListItem.slotClassNames.main} styles={styles.main}>
-            <Flex gap="gap.smaller">
-              <Flex.Item grow>{headerElement}</Flex.Item>
-              {headerMediaElement}
-            </Flex>
-            <Flex gap="gap.smaller">
-              <Flex.Item grow>{contentElement}</Flex.Item>
-              {contentMediaElement}
-            </Flex>
+          <Flex column={hasBothParts} className={ListItem.slotClassNames.main} styles={styles.main}>
+            {this.wrapWithFlex(headerPart, hasBothParts)}
```

Comment:
Sorry for commenting after this was merged. Why we have confition for adding Flex if the content and header parts exists, when the Flex we are using are adding gap between the element + mediaElement. With this we have regression (the gap between header + headerMedia and content + contentMedia are incorrect if the header and content are not applied both..).

Previous:
![image](https://user-images.githubusercontent.com/4512430/56118946-555fec80-5f6b-11e9-9d30-a33cd9b1559d.png)

Current:
![image](https://user-images.githubusercontent.com/4512430/56119001-76c0d880-5f6b-11e9-8690-20467458f567.png)